### PR TITLE
[Feature] Certificates

### PR DESCRIPTION
**Pull request by IMilja**
## Description
This PR implements the table and form which enables the user to upload a certificate of two types. We've implemented an uploader for files for the certificate and tied it together with both the certificate form and table.

## Change summary
- Added react-filepod library
- Setup a new layout for the profile page (we use a `layout.tsx` now)
- Created a form creating certificates
- Created a table for showing certificates

## Note
There is a potential issue with react-filepod which I need to investigate since it ignores the `acceptedFileTypes` property.

## Checklist
- [X] I have performed a self-review of my code
- [X] I tested locally
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New package or package update introduced

### CODE REVIEW

**Comment by IMilja** [comment]
@aarbanas Could you do a quick scan/review of the draft PR for now? I yet need to implement the feature for admins to be able to upload certificates but I'm gonna most likely reuse the same compontent.

File: src/app/profile/certificates/components/certificateTable/CertificateTable.tsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  deleteCertificate,
+  downloadCertificate,
+  fetchCertificates,
+} from '../../services/certificates-service';
+import { useUserSession } from '@/services/auth/useUserSession';
+import { type CertificateModel } from '../../models/certificate.model';
+import { CertificateTypeEnum } from '../../enums/certificate-types.enum';
+import { CERTIFICATE_TRANSLATION } from '../../constants/certificate-translation';
+
+export const resolveIcon = (type: CertificateTypeEnum) => {
```
Comment (aarbanas) [inline src/app/profile/certificates/components/certificateTable/CertificateTable.tsx:25]: Do we need to export it or is it just local variable?

File: src/app/profile/certificates/components/certificateTable/CertificateTable.tsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+      await deleteCertificate(certificate.id);
+      setCertificates(certificates.filter((c) => c.id !== certificate.id));
+    },
+    [certificates]
+  );
+
+  const handleFetchCertificates = useCallback(async (id: number) => {
+    const certificates = await fetchCertificates(id);
+    setCertificates(certificates);
+  }, []);
```
Comment (aarbanas) [inline src/app/profile/certificates/components/certificateTable/CertificateTable.tsx:60]: Do you think wrapping it to `useCallback` is a little over-engineering? Can we just call `fetchCertificates` in `useEffect`?

File: src/app/profile/certificates/models/certificate.model.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { CertificateTypeEnum } from '../enums/certificate-types.enum';
+
+export interface CertificateModel {
```
Comment (aarbanas) [inline src/app/profile/certificates/models/certificate.model.ts:3]: Is it a `CertificateModel` or `CertificateDTO`?

**Review by aarbanas — COMMENTED** [review]
Great job 🚀 
Leaving some small comments up for discussion

File: src/app/profile/certificates/components/certificateTable/CertificateTable.tsx (hunk 1 above)
Comment (IMilja) [reply src/app/profile/certificates/components/certificateTable/CertificateTable.tsx:28]: We do not need to export it. Muscle memory to export everything😅

File: src/app/profile/certificates/models/certificate.model.ts (hunk 3 above)
Comment (IMilja) [reply src/app/profile/certificates/models/certificate.model.ts]: I like Dto more so we're gonna got with that thanks!

**Comment by IMilja** [comment]
@aarbanas I left out the showing of certificates on each profile page and I'm gonna do it in a separate PR. This PR already has enough of file changes :)

**Comment by IMilja** [comment]
@aarbanas I've dropped a new design for the /user-profile page where we show user details and certificates in this PR as well.
image

**Comment by aarbanas** [comment]
> @aarbanas I've dropped a new design for the /user-profile page where we show user details and certificates in this PR as well. image

I love it 🚀

**Review by aarbanas — APPROVED** [review]

**Pull request merged**